packages/data/src/components/use-dispatch/test/use-dispatch.js

```diff
@@ -134,22 +130,17 @@ describe( 'useDispatch', () => {
 			return <button onClick={ () => dispatch( 'demo' ).noop() } />;
 		};
 
-		let testRenderer;
-		act( () => {
-			testRenderer = TestRenderer.create(
-				<RegistryProvider value={ firstRegistry }>
-					<TestComponent />
-				</RegistryProvider>
-			);
-		} );
-		const testInstance = testRenderer.root;
-
-		act( () => {
-			testInstance.findByType( 'button' ).props.onClick();
-		} );
+		const { rerender } = render(
+			<RegistryProvider value={ firstRegistry }>
+				<TestComponent />
+			</RegistryProvider>
+		);
 
+		await user.click( screen.getByRole( 'button' ) );
 		expect( firstRegistryAction ).toHaveBeenCalledTimes( 1 );
-		expect( secondRegistryAction ).toHaveBeenCalledTimes( 0 );
+		expect( secondRegistryAction ).not.toHaveBeenCalled();
+
+		firstRegistryAction.mockClear();
```

Comment:
I liked the original code more: without clearing the mocks, just checking that the number of calls goes from `(0, 0)` to `(1, 0)` and finally to `(1, 1)`.

Reply:
Do we want to keep at least one of these "called times" checks or entirely switch to asserting the store changes?

Reply:
That's what I figure, but wanted to be sure :)